crates/ruff_linter/src/settings/types.rs

```diff
@@ -647,6 +647,25 @@ where
     }
 }
 
+/// Call [`Glob::new`] but peek at the [`globset::ErrorKind`] and try the glob again with any nested
+/// alternates escaped.
+///
+/// For example, the `cookiecutter` package generates paths like `{{cookiecutter.repo_name}}`, which
+/// look like nested glob alternate expressions like `{foo,bar}`. The nested version is not
+/// supported by `globset`, causing an error.
+pub fn try_glob_new(s: &str) -> Result<Glob> {
+    match Glob::new(s) {
+        Err(e) if *e.kind() == globset::ErrorKind::NestedAlternates => {
+            let new = s.replace("{{", "[{][{]").replace("}}", "[}][}]");
```

Comment:
This is also not _technically_ right in all cases I think, e.g., `[{{]`, although that's pretty pathological.